infrastructure/common-images/workspaces-base/backend/api/helpers/density_map/generate_image.py

```diff
@@ -88,3 +81,20 @@ def generate_image_contour(scaled_img_array: np.array, dashed=False) -> (np.arra
                 contour_img_array[row_up][col_left] = 0
 
     return contour_img_array, black_to_transparent(get_image_from_array(contour_img_array, 'RGB'))
+
+
+def generate_canal_image(bg_atlas: ICustomAtlas, subdivision: str) -> (np.array, Image):
+    # TODO: Remove following line when no longer in need of mocked data
+    canal_data = np.repeat(_get_img_array(bg_atlas, 'C1-Rostral', bg_atlas.canal)[np.newaxis,:, :], 790, axis=0)
+    return _generate_image(bg_atlas, subdivision, canal_data, GREY_SCALE_MAX_CANAL, CANAL_IMAGE_OPACITY)
+
+
+def generate_grid_image(bg_atlas: ICustomAtlas, subdivision: str) -> (np.array, Image):
+    # TODO: Remove following line when no longer in need of mocked data
+    canal_data = np.repeat(_get_img_array(bg_atlas, 'C1-Rostral', bg_atlas.canal)[np.newaxis,:, :], 790, axis=0)
```

Comment:
generate canal image:
- gets the canal image array
- converts image array to image

generate_grid_image only needs the canal image array so I thought it would be better to skip the unnecessary 'converts image array to image' step that would come by calling the generate canal image.

Reply:
@afonsobspinto then may be you should refactor out the get canal image array from the generate canal image function so that you can use that new function

Reply:
That's already in place no? Maybe I miss understood your original comment. I will ping you on slack after lunch and we check this through a quick call if you are available